micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -31,9 +32,28 @@ public class MicrometerMetrics implements Metrics {
   private static final String RECONCILIATIONS_QUEUE_SIZE = PREFIX + RECONCILIATIONS + "queue.size.";
   private final MeterRegistry registry;
   private final Map<String, AtomicInteger> gauges = new ConcurrentHashMap<>();
+  private final Map<ResourceID, Set<Meter.Id>> metersPerResource = new ConcurrentHashMap<>();
+  private final Cleaner cleaner;
 
   public MicrometerMetrics(MeterRegistry registry) {
+    this(registry, 0);
+  }
+
+  public MicrometerMetrics(MeterRegistry registry, int cleanUpDelayInSeconds) {
+    this(registry, cleanUpDelayInSeconds, 0);
```

Comment:
I think the cleaning thread number should be by default 1, it is basically a very simple operation should not eath up more threads.

Reply:
Makes sense. The default (not obvious from just looking at this line) is actually the number of available processors but will make it 1.

Reply:
Done. Also added javadoc for the constructors.